code/game/objects/items/devices/minimap_tablet.dm

```diff
@@ -0,0 +1,304 @@
+		drawn_image.icon = mona_lisa
+		return
+	if(start_y == end_y)
+		drawn_image.icon = mona_lisa
+		mona_lisa.DrawBox(draw_color, start_x*2, start_y*2, end_x*2 + 1, start_y*2 + 1)
+		return
+
+	mona_lisa.DrawBox(draw_color, start_x*2, start_y*2, start_x*2 + 1, start_y*2 + 1)
+
+	var/abs_dx = abs(end_x - start_x)
+	var/abs_dy = abs(end_y - start_y)
+	var/sign_dx = SIGN(end_x - start_x)
+	var/sign_dy = SIGN(end_y - start_y)
+
+	//special case 2, perfectly diagonal line
+	if(abs_dx == abs_dy)
+		for(var/j = 1 to abs_dx)
+			start_x += sign_dx
+			start_y += sign_dy
+			mona_lisa.DrawBox(draw_color, start_x*2, start_y*2, start_x*2 + 1, start_y*2 + 1)
+		drawn_image.icon = mona_lisa
+		return
+
+	/*x_error and y_error represents how far we are from the ideal line.
+	Initialized so that we will check these errors against 0, instead of 0.5 * abs_(dx/dy)*/
+	//We multiply every check by the line slope denominator so that we only handles integers
+	if(abs_dx > abs_dy)
+		var/y_error = -(abs_dx >> 1)
+		var/steps = abs_dx
+		while(steps--)
+			y_error += abs_dy
+			if(y_error > 0)
+				y_error -= abs_dx
+				start_y += sign_dy
+			start_x += sign_dx
+			mona_lisa.DrawBox(draw_color, start_x*2, start_y*2, start_x*2 + 1, start_y*2 + 1)
+	else
+		var/x_error = -(abs_dy >> 1)
+		var/steps = abs_dy
+		while(steps--)
+			x_error += abs_dx
+			if(x_error > 0)
+				x_error -= abs_dy
+				start_x += sign_dx
+			start_y += sign_dy
+			mona_lisa.DrawBox(draw_color, start_x*2, start_y*2, start_x*2 + 1, start_y*2 + 1)
+	drawn_image.icon = mona_lisa
+
+/atom/movable/screen/minimap_tool/draw_tool/red
+	screen_loc = "16,14"
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/draw_red.dmi'
+	color = MINIMAP_DRAWING_RED
+
+/atom/movable/screen/minimap_tool/draw_tool/yellow
+	screen_loc = "16,13"
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/draw_yellow.dmi'
+	color = MINIMAP_DRAWING_YELLOW
+
+/atom/movable/screen/minimap_tool/draw_tool/purple
+	screen_loc = "16,12"
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/draw_purple.dmi'
+	color = MINIMAP_DRAWING_PURPLE
+
+/atom/movable/screen/minimap_tool/draw_tool/blue
+	screen_loc = "16,11"
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/draw_blue.dmi'
+	color = MINIMAP_DRAWING_BLUE
+
+/atom/movable/screen/minimap_tool/draw_tool/erase
+	icon_state = "erase"
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/draw_erase.dmi'
+	screen_loc = "16,10"
+	color = null
+
+/atom/movable/screen/minimap_tool/label
+	icon_state = "label"
+	active_mouse_icon = 'icons/UI_Icons/minimap_mouse/label.dmi'
+	screen_loc = "16,8"
+	/// List of turfs that have labels attached to them. kept around so it can be cleared
+	var/list/turf/labelled_turfs = list()
+
+/atom/movable/screen/minimap_tool/label/Click(location, control, params)
+	. = ..()
+	var/list/modifiers = params2list(params)
+	if(modifiers[BUTTON] == RIGHT_CLICK)
+		log_minimap_drawing("[key_name(usr)] has cleared current labels")
```

Comment:
could this be made to work so that if an aghosted admin does this they can delete it too?

Reply:
could yes but I dont feel like doing it in this pr